### PR TITLE
add autoNetworkTier to Router NAT

### PR DESCRIPTION
**Pull request by maxi-cit**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/16139

It adds support to set the network tier to use when automatically reserving NAT IP Addresses

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `auto_network_tier` field to `google_compute_router_nat` resource
```

### CODE REVIEW

File: mmv1/products/compute/RouterNat.yaml [hunk 1]
```diff
@@ -436,3 +436,13 @@ properties:
...
+  - !ruby/object:Api::Type::Enum
+    name: 'autoNetworkTier'
+    required: false
+    description: |
+      The network tier to use when automatically reserving NAT IP addresses.
+      Must be one of: PREMIUM, STANDARD. If not specified, then the current
+      project-level default tier is used.
+    values:
+      - :PREMIUN
+      - :STANDAR
```
Comment (trodge) [inline mmv1/products/compute/RouterNat.yaml:448]:
```suggestion
      - :STANDARD
```

**Review by trodge — CHANGES_REQUESTED** [review]
We will need some way to handle the values the API returns for this field for existing resources.

The field value is derived from a setting on the project that the resource belongs to, so one option would be to try to look up that value and set it in an encoder when it's unspecified. However, I don't know of an easy way to get that information.

The alternative would be to use `default_from_api: true` on this field, but that would mean that deleting this field from the configuration would preserve whatever the last value was even if the project-level default changed.

**Comment by maxi-cit** [comment]
Hello @trodge, I have been studying the first case and I think is somewhat doable.
However, could you explain about the latter case, please?

File: mmv1/products/compute/RouterNat.yaml (hunk 1 above)
Comment (maxi-cit) [reply mmv1/products/compute/RouterNat.yaml]: done

**Comment by trodge** [comment]
> Hello @trodge, I have been studying the first case and I think is somewhat doable. However, could you explain about the latter case, please?

Setting `default_from_api` will make the field `Optional` and `Computed` in the provider, which means that any value returned from the API will be treated as though it matches an empty value in the user's config.

**Comment by maxi-cit** [comment]
Helllo @trodge I tried to implement your encoder fix so it will take the project default when empty. Could you take a look, please?

**Comment by melinath** [comment]
@maxi-cit is this PR still active?

**Comment by maxi-cit** [comment]
Hello @melinath, sorry for the late reply. it is active but other products got higher prio. However I will comeback to this one later in the week.

**Comment by maxi-cit** [comment]
Hello @trodge I updated this PR making this new field default_from_api. I tried the encoder fix but it does not seems that this field should be tied to the project level default changes. I believe that we can have this field 'immutable' so it gets recreated anytime the API returns a different network tier value.
do you think I should add a special note for this field?

**Review by trodge — CHANGES_REQUESTED** [review]
It looks like `testAccComputeRouterNatBaseResourcesWithPrivateNatSubnetworks` was moved inside `<% unless version == "ga" -%>` guards, making it beta only. To get this to compile you'll need to move that function outside the beta-only guard.

**Comment by maxi-cit** [comment]
Hello @trodge  thanks for the feedback. I applied your suggestions and moved the function. This should be fixing it, but if fails on the ga-provider I will make a separate one

**Review by trodge — APPROVED** [review]

**Pull request merged**